src/sema/types.rs

```diff
@@ -1393,17 +1508,12 @@ impl Type {
                     .definition(ns)
                     .fields
                     .iter()
-                    .map(|f| {
-                        if f.recursive {
-                            BigInt::one()
-                        } else {
-                            f.ty.storage_slots(ns)
-                        }
-                    })
+                    .filter(|f| !f.infinite_size)
+                    .map(|f| f.ty.storage_slots(ns))
                     .sum(),
+                Type::Array(_, dims) if dims.last() == Some(&ArrayLength::Dynamic) => 1.into(),
                 Type::Array(ty, dims) => {
-                    let one = BigInt::one();
-
+                    let one = 1.into();
```

Comment:
I still prefer the old notation. It makes it clear we have a BigInt.

Reply:
I do not. It is already clear we have a BigInt from the function signature. My version is less to type. I find my version easier to read. So in my opinion, it is better like that. Bike shedding :smile:

Reply:
No problem to leave it there. I left a comment because this was a random change in the codebase - it was not related to this PR.

Reply:
True, looks random, I must have touched this at one point while working on something else without even noticing